dsl_parser/tests/test_functions.py

```diff
@@ -1251,134 +1250,6 @@ def test_func_integrations_within_list(self):
         some_node = self.get_node_by_name(parsed, 'some_node')
         self.assertEqual(['42'], some_node['properties']['concat_prop'])
 
-    def test_get_input_doesnt_accept_runtime_func_as_args(self):
-        yaml = """
-node_types:
-    some_type:
-        properties:
-            prop: {}
-            dummy_prop: {}
-node_templates:
-    some_node:
-        type: some_type
-        properties:
-            dummy_prop: dummy_value
-            prop:
-                get_input:
-                    - get_attribute:
-                            - some_node
-                            - dummy_prop
-    """
-        with self.assertRaisesRegexp(
-                exceptions.FunctionValidationError,
-                'Runtime function .+ cannot be nested within a non-runtime '
-                'function \\(found in .+\\)'):
-            prepare_deployment_plan(self.parse(yaml))
-
-        yaml = """
-node_types:
-    some_type:
-        properties:
-            prop: {}
-            dummy_prop: {}
-node_templates:
-    some_node:
-        type: some_type
-        properties:
-            dummy_prop: dummy_value
-            prop:
-                get_input:
-                    concat:
-                        - get_attribute:
-                                - some_node
-                                - dummy_prop
-    """
-        with self.assertRaisesRegexp(
-                exceptions.FunctionValidationError,
-                'Runtime function .+ cannot be nested within a non-runtime '
-                'function \\(found in .+\\)'):
-            prepare_deployment_plan(self.parse_1_1(yaml))
-
-        yaml = """
-inputs:
-    dummy_input:
-        default: some_node
-node_types:
-    some_type:
-        properties:
-            prop: {}
-            dummy_prop: {}
-node_templates:
-    some_node:
-        type: some_type
-        properties:
-            dummy_prop: dummy_value
-            prop:
-                get_attribute:
-                    - get_input: dummy_input
-                    - concat:
-                        - get_attribute: [ some_node, dummy_prop ]
-                    - get_input:
-                        concat:
-                            - get_attribute: [ some_node, dummy_prop ]
-                            - get_input: dummy_input
-    """
-        with self.assertRaisesRegexp(
-                exceptions.FunctionValidationError,
-                'Runtime function .+ cannot be nested within a non-runtime '
-                'function \\(found in .+\\)'):
-            prepare_deployment_plan(self.parse_1_1(yaml))
-
-    def test_get_property_doesnt_accept_runtime_func_as_args(self):
```

Comment:
so what about tests that test this works :)

Reply:
re-added that validation and added a test

old test did `prepare_deployment_plan(parse())` but it actually threw the error inside of parse, so the prepare was useless

however now the error is thrown inside of prepare, because it is allowed now to have functions nested like that, it will only throw in prepare (creating a deployment), not parse (uploading a blueprint)

so thats a change in behaviour in this pr